contracts/NodeDelegatorETH.sol

```diff
@@ -0,0 +1,411 @@
+    /// @param publicKeys Array of validator public keys
+    /// @param operatorIds The operator IDs of the SSV Cluster
+    function exitSsvValidators(
+        bytes[] calldata publicKeys,
+        uint64[] calldata operatorIds
+    )
+        external
+        onlyLRTOperator
+        whenNotPaused
+    {
+        address SSV_NETWORK_ADDRESS = lrtConfig.getContract(LRTConstants.SSV_NETWORK);
+
+        ISSVNetwork(SSV_NETWORK_ADDRESS).bulkExitValidator(publicKeys, operatorIds);
+    }
+
+    /// @dev Remove validators from the SSV Cluster.
+    /// Make sure `exitSsvValidators` is called before and all the validators have exited the Beacon chain.
+    /// If removed before the validator has exited the beacon chain will result in the validator being slashed.
+    /// Only the Operator can call this function.
+    /// @param publicKeys Array of validator public keys
+    /// @param operatorIds The operator IDs of the SSV Cluster
+    /// @param cluster The SSV cluster details including the validator count and SSV balance
+    function removeSsvValidators(
+        bytes[] calldata publicKeys,
+        uint64[] calldata operatorIds,
+        Cluster calldata cluster
+    )
+        external
+        onlyLRTOperator
+        whenNotPaused
+    {
+        address SSV_NETWORK_ADDRESS = lrtConfig.getContract(LRTConstants.SSV_NETWORK);
+
+        ISSVNetwork(SSV_NETWORK_ADDRESS).bulkRemoveValidator(publicKeys, operatorIds, cluster);
+    }
+
+    /// @dev Withdraw all ether in the EigenPod which can be from beacon consensus rewards or validator exits.
+    /// The ether will be sent to the EigenLayer's DelayedWithdrawalRouter contract where it can be claimed after seven
+    /// days.
+    function requestEthWithdrawal() external onlyLRTOperator {
+        if (eigenPod == address(0)) {
+            revert NoEigenPod();
+        }
+        IEigenPod(eigenPod).withdrawBeforeRestaking();
+    }
+
+    /// @dev Claim previously requested ether withdrawals from EigenLayer's DelayedWithdrawalRouter contract.
+    /// Need to account if the ether is from validator exits or beacon chain consensus rewards.
+    function claimEthWithdrawal() external onlyLRTOperator {
+        uint256 ethBefore = address(this).balance;
+
+        address delayerWithdrawalRouter = lrtConfig.getContract(LRTConstants.EIGEN_DELAYED_WITHDRAWAL_ROUTER);
+        // Only claim one withdrawal request at a time to make the accounting easier
+        IDelayedWithdrawalRouter(delayerWithdrawalRouter).claimDelayedWithdrawals(1);
+
+        uint256 ethClaimed = address(this).balance - ethBefore;
+
+        uint256 fullyWithdrawnValidators;
+        // Account for the claimed ether
+        if (ethClaimed >= FULL_STAKE) {
+            // explicitly cast to uint256 as we want to round to a whole number of validators
+            fullyWithdrawnValidators = uint256(ethClaimed / FULL_STAKE);
+            stakedButNotVerifiedEth -= fullyWithdrawnValidators * FULL_STAKE;
```

Comment:
🟥In the event that we have earned more than 32 ETH from our validators over time, this line would crash, correct?

Maybe okay since we are doing this withdraw as a one time thing, but it should probably at least get a comment.